### PR TITLE
Fix LinkTo usage and explicitly provide `@route` and `@models` arguments

### PR DESCRIPTION
**Pull request by SergeAstapov**

### CODE REVIEW

File: addon/components/api/x-section/index.js [hunk 1]
```diff
@@ -2,16 +2,14 @@ import Component from '@ember/component';
...
   @class Api/XSection
   @hide
 */
-export default Component.extend({
-  tagName: '',
+export default class ApiXSection extends Component {
+  tagName = '';
 
-  showImportPaths,
+  showImportPaths = config['ember-cli-addon-docs'];
```
Comment (rwwagner90) [inline addon/components/api/x-section/index.js:12]: This does not seem correct. `showImportPaths` was destructured before.

File: addon/components/docs-header/search-result/index.hbs [hunk 2]
```diff
@@ -1,12 +1,12 @@
-{{! template-lint-disable no-unknown-arguments-for-builtin-components }}
 <div {{on 'click' this.onClick}} {{on 'mouseenter' this.onMouseEnter}} data-test-search-result>
   <LinkTo
     class={{
       concat
       'docs-block docs-py-2 docs-px-3 docs-text-black docs-no-underline hover:docs-bg-grey-lighter '
       (if @selected 'docs-bg-grey-lighter')
     }}
-    @params={{this.linkArgs}}
+    @route={{this.linkArgs.route}}
```
Comment (rwwagner90) [inline addon/components/docs-header/search-result/index.hbs:8]: I think I had split these args out like this before and it broke things. Are you sure this works?

File: addon/components/docs-viewer/x-main/index.hbs [hunk 3]
```diff
@@ -41,7 +42,8 @@
       <LinkTo
         class='docs-text-grey-darkest docs-text-large-4 docs-font-light docs-no-underline
         docs-border-b docs-border-grey hover:docs-border-grey-darkest docs-transition'
-        @params={{this.docsRoutes.next.route}}
+        @route={{this.docsRoutes.next.route}}
```
Comment (rwwagner90) [inline addon/components/docs-viewer/x-main/index.hbs:45]: Same with these

**Review by rwwagner90 — CHANGES_REQUESTED** [review]

File: addon/components/api/x-section/index.js (hunk 1 above)
Comment (SergeAstapov) [reply addon/components/api/x-section/index.js:12]: @rwwagner90 ah yes, good catch!
Actually I'll revert this change, I planned to make native class conversion in separate PR to make things focused

Comment (SergeAstapov) [reply addon/components/api/x-section/index.js]: reverted!

File: addon/components/docs-header/search-result/index.hbs (hunk 2 above)
Comment (SergeAstapov) [reply addon/components/docs-header/search-result/index.hbs:8]: @rwwagner90 yes, this works in combination with `@models` argument

File: addon/components/docs-viewer/x-main/index.hbs (hunk 3 above)
Comment (SergeAstapov) [reply addon/components/docs-viewer/x-main/index.hbs:45]: @rwwagner90 yes, this works in combination with `@models` argument

**Pull request merged**